comptests/prov/test_prov_k8s_cortx_cold_upgrade.py

```diff
@@ -0,0 +1,210 @@
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Provisioner Component level test cases for K8s CORTX Software Cold Upgrade."""
+
+import logging
+import os
+import time
+import queue
+from threading import Thread
+import pytest
+
+from commons import commands
+from commons import constants as cons
+from commons.helpers.pods_helper import LogicalNode
+from commons.utils import assert_utils
+from config import CMN_CFG, PROV_CFG, PROV_TEST_CFG
+from libs.ha.ha_common_libs_k8s import HAK8s
+from libs.prov.prov_k8s_cortx_deploy import ProvDeployK8sCortxLib
+
+LOGGER = logging.getLogger(__name__)
+
+
+class TestProvK8CortxColdUpgrade:
+    """
+    This class contains Provisioner Component level test cases for K8s
+    CORTX Software Cold Upgrade.
+    """
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class"""
+        LOGGER.info("STARTED: Setup Module operations")
+        cls.cortx_all_image = os.getenv("CORTX_ALL_IMAGE", None)
+        cls.cortx_rgw_image = os.getenv("CORTX_RGW_IMAGE", None)   
+        cls.deploy_cfg = PROV_CFG["k8s_cortx_deploy"]
+        cls.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.worker_node_list = []
+        cls.master_node_list = []
+        cls.host_list = []
+        cls.local_sol_path = cons.LOCAL_SOLUTION_PATH
+        for node in range(cls.num_nodes):
+            node_obj = LogicalNode(hostname=CMN_CFG["nodes"][node]["hostname"],
+                                   username=CMN_CFG["nodes"][node]["username"],
+                                   password=CMN_CFG["nodes"][node]["password"])
+            cls.host_list.append(node_obj)
+            if CMN_CFG["nodes"][node]["node_type"].lower() == "master":
+                cls.master_node_obj = node_obj
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        LOGGER.info("Done: Setup operations finished.")
+
+    def perform_upgrade(self, exc: bool = True, output=None):
+        """Function calls upgrade and put return value in queue object."""
+        LOGGER.info("Calling upgrade.")
+        resp = self.deploy_lc_obj.upgrade_software(self.master_node_obj,
+                                                   self.prov_deploy_cfg["git_remote_path"],
+                                                   exc=exc)
+        output.put(resp)
+
+    @pytest.mark.run(order=1)
+    @pytest.mark.lc
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37357")
+    def test_37357(self):
+        """
+        Verify CORTX Software Cold upgrade.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Step 1: Get installed version.")
```

Comment:
Please add a library function to retrieve installed version . (Step 1)

Same can be done for comparing two version. (Step 2)